### ISSUE TITLE
Well-known symbols enumerability

### ISSUE THREAD

**Issue by zloirock** [babel/babel#1922]
Symbols can't be fully pofyfilled. For most symbols `core-js` uses setters in `Object.prototype`, but it breaks `Symbol() in {}`. For well-known symbols something like `Symbol.iterator in foo` is very often case, because this I don't add setters for well-known symbols to `Object.prototype`. It breaks non-enumerability for them. I did not think that it is a serious problem, but now I see this case in some projects, for example, see `for-of-object` test for IE11 and Safari [here](http://kpdecker.github.io/six-speed/). I can't fully solve this problem in polyfill side. Possible sollution - transform setting well-known symbols:
```js
foo[Symbol.something] = bar;
// =>
Object.defineProperty(foo, Symbol.something, {value: bar, enumerable: true, configurable: true, writable: true});
```
`Object.defineProperty` and other methods wrapped for correct work in this case.

**Comment by RReverser** [comment]
Note that such transform won't work 100% correct as we can't track all the cases of `foo[someIndex] = bar;` where `someIndex` might be a `Symbol`.

**Comment by zloirock** [comment]
@RReverser this transform required only for well-known symbols like `Symbol.iterator`. I don't think something like:
```js
var ITERATOR = Symbol.iterator;
foo[ITERATOR] = fn;
```
is very often case. Anyway, it looks better than break `Symbol.iterator in foo`.

**Comment by RReverser** [comment]
Maybe. Just suggesting alternative solution that would cover both, but, whatever works best :)

**Comment by zloirock** [comment]
@RReverser yep, transform `x in y` interesting solution. Need think about it.
